// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto3";

package optimization_guide.proto;

import "components/optimization_guide/proto/features/common_quality_data.proto";

option optimize_for = LITE_RUNTIME;
option java_package = "org.chromium.components.optimization_guide.features.proto";

option java_outer_classname = "EnhancedCalendarProto";

// DO NOT EDIT THIS FILE DIRECTLY!
//
// This file is generated in g3 and then synced to Chrome. Instead, please
// refer to http://go/chrome-intelligence-feature-protos (Google-internal link),
// and then changes will be synced with Chrome automatically.

message EnhancedCalendarLoggingData {
  // The feature request.
  EnhancedCalendarRequest request = 1;

  // The feature response.
  EnhancedCalendarResponse response = 2;
}

// The request to execute an Enhanced Calendar query.
// Next ID: 5
message EnhancedCalendarRequest {
  // The start date of the event. Ideally in the format of HH:MM MM/DD/YYYY
  // (E.g. 10:00 AM 01/01/2024) but because this is an LLM output, the format is
  // not guaranteed.
  string start_date = 1;

  // The end date of the event. Ideally in the format of HH:MM MM/DD/YYYY (E.g.
  // 10:00 AM 01/01/2024) but because this is an LLM output, the format is not
  // guaranteed.
  string end_date = 2;

  // The web page context used for generating the event. Context is expected to
  // contain calendar event dates or related information to populate the
  // calendar description.
  PageContext page_context = 3;

  // The prompt used for generating the event. If the prompt is not provided,
  // the default prompt from the feature config will be used.
  string prompt = 4;
}

// The response from executing a Enhanced Calendar query.
// Next ID: 8
message EnhancedCalendarResponse {
  // The model's output title for the calendar event.
  string event_title = 1;

  // The model's output summary for the calendar event.
  string event_summary = 2;

  // The model's output start date for the calendar event.
  string start_date = 3;

  // The model's output end date for the calendar event.
  string end_date = 4;

  // The model's output recurrence state for the calendar event.
  RecurrenceState recurrence = 5;

  // The model's output event location for the calendar event.
  string event_location = 6;

  // The model's output for if the event is an all day event.
  bool is_all_day = 7;
}

// The recurrence cadence for a repeated calendar event. If there's no
// reoccurrence or there's a reoccurrence that isn't strictly one of the enum
// options, state "STATE_UNSPECIFIED" should be used. Next ID: 6
enum RecurrenceState {
  // An event that is not repeated.
  STATE_UNSPECIFIED = 0;

  // An event that repeats every day.
  STATE_EVERY_DAY = 1;

  // An event that repeats every week.
  STATE_EVERY_WEEK = 2;

  // An event that repeats every two weeks.
  STATE_EVERY_TWO_WEEKS = 3;

  // An event that repeats every month.
  STATE_EVERY_MONTH = 4;

  // An event that repeats every year.
  STATE_EVERY_YEAR = 5;
}
